core/core-impl/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/MoveSelectorConfig.java

```diff
@@ -280,4 +282,33 @@ private void inheritCommon(MoveSelectorConfig<?> inheritedConfig) {
                 fixedProbabilityWeight, inheritedConfig.getFixedProbabilityWeight());
     }
 
+    protected String addRandomSuffix(String name, Random random) {
+        StringBuilder value = new StringBuilder(name);
+        value.append("-");
+        random.ints(97, 122) // ['a', 'z']
+                .limit(4) // 4 letters
+                .forEach(value::appendCodePoint);
+        return value.toString();
+    }
+
+    /**
+     * Check if the selector factory accepts Nearby selection autoconfiguration.
+     *
+     * @return false by default
+     */
+    public boolean acceptNearbySelectionAutoConfiguration() {
+        return false;
+    }
+
+    /**
+     * Enables the Nearby Selection autoconfiguration.
+     *
+     * @return new instance with the Nearby Selection settings properly configured
+     */
+    public Config_ enableNearbySelection(Class<? extends NearbyDistanceMeter<?, ?>> distanceMeter, Random random) {
+        throw new UnsupportedOperationException();
+    }
+
+    public abstract boolean hasNearbySelectionConfig();
```

Comment:
Maybe this could return `false` by default, so that you only need to override it when `true` needs to be returned?